packages/@react-aria/utils/src/useId.ts

```diff
@@ -89,7 +89,7 @@ export function useSlotId(depArray: ReadonlyArray<any> = []): string {
     setResolvedId(function *() {
       yield id;
 
-      yield document.getElementById(id) ? id : null;
+      yield document.getElementById(id) ? id : undefined;
```

Comment:
These changes from null to undefined are to better accommodate mergeProps, which treats undefined differently from null. If null, this will override the previous value, whereas undefined will not.